### PR TITLE
1.7.0

### LINKED ISSUE #128: [NFR] Use Phalcon binary package/dll
To speed up the action when using the Phalcon extension it would be great if there's an option to use the binary package on linux or precompiled dll's on Windows. 

Windows dll's can be found here.
https://github.com/phalcon/cphalcon/releases/tag/v4.0.0

Linux 
https://docs.phalcon.io/4.0/en/installation#linux

### LINKED ISSUE #132: Windows - PHP 7.4.1 not available on Windows
**Describe the bug**
PHP 7.4.1 is not available on Windows. Working on Unbuntu.

**To Reproduce**
https://github.com/julien-boudry/Condorcet/blob/dev-2.1/.github/workflows/execute_all_tests.yml

**Additional context**
7.4.1 fixes many important bugs, especially on Windows (Notably the fact that interactive shells don't work at all).

### PR DESCRIPTION
**Pull request by shivammathur**
- Refactor installation scripts.
  - Switching between installed `PHP` versions should now just take seconds on `linux`. 🚀
  - Plugin `hirak/prestissimo` is now optional, can be setup using `tools` input.
  - Update `ppa:ondrej/php` only if required.
- Input `pecl` is now deprecated, specify `pecl` in `tools` input.
- Fix regex to check installed extensions in `linux` and `darwin`.
- Discard patch if specified in `php-version` input.
- Add cross-platform support for `ext-phalcon` using pre-built binaries. Fixes #128 
- Refactor coverage and extension code.
- Add support to setup widely used tools globally.
- Update `symfony` examples
- Update existing PHP version on windows. Fixes #132

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,14 +226,20 @@ You can persist composer's internal cache directory using the [`action/cache`](h
 - name: Cache dependencies  
   uses: actions/cache@v1
   with:
-    path: ${{ steps.composer-cache.outputs.dir }}
+    path: ${{ steps.composer-cache.outputs.dir }}
```
Comment (drupol) [inline README.md]: Unrelated change

Comment (shivammathur) [reply README.md]: Fixed in 72e4730
